src/openssl_wrap.cpp

```diff
@@ -864,6 +864,46 @@ std::string _X509_NAME_ENTRY_get_data(X509_NAME_ENTRY *entry)
     return bio.flushToString();
 }
 
+#pragma mark -- Symmetric Encryption Interface --
+
+SSL_EVP_CIPHER_CTX_Ptr _EVP_CIPHER_CTX_new() {
+    return SSL_EVP_CIPHER_CTX_Ptr{OpensslCallPtr::callChecked(lib::OpenSSLLib::SSL_EVP_CIPHER_CTX_new)};
+}
+
+
+void _EVP_CIPHER_CTX_ctrl(EVP_CIPHER_CTX *ctx, int type, int arg, void *ptr) {
+    OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_EVP_CIPHER_CTX_ctrl, ctx, type, arg, ptr);
+}
+
+void _EVP_CipherInit_ex(EVP_CIPHER_CTX *ctx,
+                        const EVP_CIPHER* cipher,
+                        ENGINE* impl,
+                        const unsigned char* key,
+                        const unsigned char* iv,
+                        int enc) {
+    OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_EVP_CipherInit_ex, ctx, cipher, impl, key, iv, enc);
+}
+
+void _EVP_CipherUpdate(EVP_CIPHER_CTX * ctx,
+                       unsigned char * out,
+                       int * outl,
+                       const unsigned char * in,
+                       int inl) {
+    OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_EVP_CipherUpdate, ctx, out, outl, in, inl);
+}
+
+void _EVP_CipherFinal_ex(EVP_CIPHER_CTX* ctx, unsigned char* outm, int* outl) {
+    OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_EVP_CipherFinal_ex, ctx, outm, outl);
+}
+
+int _EVP_CIPHER_CTX_key_length(const EVP_CIPHER_CTX *ctx) {
+    return lib::OpenSSLLib::SSL_EVP_CIPHER_CTX_key_length(ctx);
```

Comment:
Thanks. fixed.

Reply:
Inside OpenSSL it is implemented as 
```cpp
int EVP_CIPHER_CTX_key_length(const EVP_CIPHER_CTX *ctx)
{
    return ctx->key_len;
}
```

Reply:
I'm not convinced that the example above makes sense in our situation. We create a cipher with default attributes, we call EVP_CIPHER_CTX_key_length() to find out what key size is set by default and return "invalid key length" in case of error? To me it looks like a "bug" or "internal error" kind of problem.

To be on the safe side, I have added a check and throw if the returned value is < 0.

Fine with you?